projects/lights_control/src/lights_periph.c

```diff
@@ -0,0 +1,60 @@
+#include "event_queue.h"
+#include "gpio.h"
+#include "lights_events.h"
+#include "structs.h"
+
+#define NUM_PERIPH_FRONT 4
+#define NUM_PERIPH_REAR 4
+
+static GPIOAddress s_front_address_lookup[] = {
+  [EVENT_HEADLIGHTS] = { .port = 0, .pin = 0 },    //
+  [EVENT_SIGNAL_LEFT] = { .port = 0, .pin = 0 },   //
+  [EVENT_SIGNAL_RIGHT] = { .port = 0, .pin = 0 },  //
+  [EVENT_HORN] = { .port = 0, .pin = 0 }           //
+};
+
+static GPIOAddress s_rear_address_lookup[] = {
+  [EVENT_BRAKES] = { .port = 0, .pin = 0 },        //
+  [EVENT_SIGNAL_LEFT] = { .port = 0, .pin = 0 },   //
+  [EVENT_SIGNAL_RIGHT] = { .port = 0, .pin = 0 },  //
+  [EVENT_STROBE] = { .port = 0, .pin = 0 }         //
+};
+
+// same settings are used for all ports
+static GPIOSettings s_settings = { .direction = GPIO_DIR_OUT,
+                                   .state = GPIO_STATE_LOW,    // defaulted to LOW
+                                   .resistor = GPIO_RES_NONE,  // TODO(ELEC-165): figure this out
+                                   .alt_function = GPIO_ALTFN_NONE };
+
+BoardType s_boardtype;
+
+StatusCode lights_periph_init(BoardType boardtype) {
```

Comment:
If you pass in the array that you want to index into (and the length of the array), I think that's cleaner than doing these checks that you do here to determine whether or not you are `FRONT` or `BACK`.